src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -182,20 +182,22 @@ export const portfolioApi = createApi({
             .getAccount(accountSpecifier)
           const account = { [accountSpecifier]: chainAdaptersAccount }
           const data = accountToPortfolio(account)
+          baseQuery.dispatch(portfolio.actions.upsertPortfolio(data))
           return { data }
         } catch (e) {
           const status = 400
           const data = JSON.stringify(e)
           const error = { status, data }
           return { error }
         }
-      },
-      onCacheEntryAdded: async (_args, { dispatch, cacheDataLoaded, getCacheEntry }) => {
-        await cacheDataLoaded
-        const port = getCacheEntry().data
-        if (!port) return
-        dispatch(portfolio.actions.upsertPortfolio(port))
       }
+      // NOTE (amitojsingh366): Not sure if we need this now since I moved the dispatch in the query itself, this fixed #820
```

Comment:
cool, i understand why this is the fix, can you please delete this commented code?

Reply:
Sure I'll remove the commented code, do you also want me to remove the note?